### PR TITLE
Semret Nicodimos - Edges - JS-Adagrams

### PR DESCRIPTION
**Pull request by snicodimos**
# JS Adagrams
Congratulations! You're submitting your assignment!

## Comprehension Questions
Question | Answer
--- | ---
What patterns were you able to use from your Ruby knowledge to apply to JavaScript? | For the score calculation, I was able to utilize hashes. From the time we implemented the adagram in Ruby to now I am extremely comfortable using hashes and iterating through them and I was able to take that knowledge and apply it to the scoring function it to JavaScript
Did you need to use different strategies to find information online about JavaScript than you would for Ruby? | I try to utilize words that are specific to JavaScript to get better results when I was searching on google. I also heavily utilized the mdn documents for JavaScript to see what built in methods/functions are available for JavaScipt as suppose to ruby. 
What was a challenge you were able to overcome on this assignment? | Having to look over our ruby code 15 weeks into the program. There were many improvements I was able to implement as I was authoring the JavaScript code. Thus, having to think of the improvements and translate it to a new language  all at once was a bit challenging. 
What is your favorite thing about learning a new programming language? | 1. building on my skills and adding a library of languages to my skill set 2. Being able to compare and contrast with ruby 3. being able to learn the corks, strengths and weakness of the program
What is your least favorite thing about learning a new programming language? | The adjustment period. Having been accustomed to one way of writing a syntax and having to change quickly and missing some portions of the new syntax here and there. For instance the use of curly braces and semi-colons
Do you have any recommendations on how we could improve this project for the next cohort? | I think this was a great project. Having done the assignment in ruby, it gives us some time to implement the new syntax with a familiar problem and design. I think its a great way to jump into the new language.

### CODE REVIEW

**Comment by tildeee** [comment]
# JS Adagrams
## What We're Looking For

Feature | Feedback
--- | ---
**General** |
Answered comprehension questions | x
Small commits with meaningful commit messages | x
**Code Requirements** |
`drawLetters` method | x
Uses appropriate data structure to store the letter distribution | x
All tests for `drawLetters` pass | x
`usesAvailableLetters` method | x
All tests for `usesAvailableLetters` pass | x
`scoreWord` method | x
Uses appropriate data structure to store the letter scores | x
All tests for `scoreWord` pass | x
`highestScoreFrom` method | x
Appropriately handles edge cases for tie-breaking logic | x
All tests for `highestScoreFrom` pass | x
**Overall** |

Great job Semret! Overall, in this project, you probably chose my favorite approaches/solutions to the Adagrams project. Most of my comments are on very specific Javascript things, such as semicolons or functions I would choose over the other. In terms of approach of problem-solving, you did really well

Well done!

File: src/adagrams.js [hunk 1]
```diff
@@ -1,7 +1,142 @@
...
+      'Y','Y',
+      'Z'
+    ];
+
+    function genLetterPool(alphabets) {
+      // we need to initilize an empty
+      let hand = [];
+      for (let i=0, sampleCount=10; i < sampleCount; i+=1) {
+        // we use Math.random maybe a combo Math.floor
+        hand.push(alphabets[Math.floor(Math.random()*alphabets.length)]);
```
Comment (tildeee) [inline src/adagrams.js:38]: This line is doing a lot... and I think that's great! This is a great way to implement this logic

File: src/adagrams.js [hunk 2]
```diff
@@ -1,7 +1,142 @@
...
+      }
+      return hand;
+    }
```
Comment (tildeee) [inline src/adagrams.js:41]: Nice job implementing an inner function inside of a function, and calling it within it. You also did a great job remembering to explicitly use a return statement.

File: src/adagrams.js [hunk 3]
```diff
@@ -1,7 +1,142 @@
...
+
+    return genLetterPool(alphabets)
```
Comment (tildeee) [inline src/adagrams.js:43]: Don't forget your semicolon!

File: src/adagrams.js [hunk 4]
```diff
@@ -1,7 +1,142 @@
...
+  },
+
+
+  usesAvailableLetters(word, hand) {
+    let tempHand = Array.from(hand);
+
+    let inputtedWord = word.toUpperCase().split("");
```
Comment (tildeee) [inline src/adagrams.js:50]: `inputtedWord` could probably be a `const`

File: src/adagrams.js [hunk 5]
```diff
@@ -1,7 +1,142 @@
...
+
+
+    for (let i=0; i < inputtedWord.length; i+=1){
+      if (tempHand.includes(inputtedWord[i])) {
+        const index = tempHand.indexOf(inputtedWord[i])
+        delete tempHand[index];
```
Comment (tildeee) [inline src/adagrams.js:56]: Good implementation of this solution-- One suggestion I have for the JavaScript solution is: I may prefer to use `splice` to get rid of an item in an array instead of `delete`. Read here for some reasons why: https://stackoverflow.com/questions/500606/deleting-array-elements-in-javascript-delete-vs-splice/38803368

File: src/adagrams.js [hunk 6]
```diff
@@ -1,7 +1,142 @@
...
+      V:4,
+      W:4,
+      X:8,
+      Y:4,
+      Z:10,
+    };
+
+    let score = 0;
+
+    let inputtedWord = word.toUpperCase().split("");
```
Comment (tildeee) [inline src/adagrams.js:98]: `inputtedWord` could probably be a `const`

File: src/adagrams.js [hunk 7]
```diff
@@ -1,7 +1,142 @@
...
+          highestScoring.score = scoreForWord;
+          highestScoring.word = word;
+        }
+        // if both words are not 10 and if new word is less letters wins
+        else if (word.length < highestScoring.word.length && highestScoring.word.length != 10) {
+          highestScoring.score = scoreForWord;
+          highestScoring.word = word;
+        }
+      }
+    }
```
Comment (tildeee) [inline src/adagrams.js:136]: this is a really clean implementation; well done!
